File: ws20/ShoppingCenter/src/main/java/ec/edu/espe/shoppingcenter/model/Customer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.shoppingcenter.model;

import java.util.ArrayList;


/**
 *
 * @author Odalys Chavez, CodeBreakers, @ESPE
 */
public class Customer {

    private int id;
    private String firstName;
    private String lastName;
    private String gender;
    private float moneySpent;
    private int age;
    private String typeOfCustomer;
    private ArrayList<String> hobbies;
    
    public Customer(int id, String firstName, String lastName, String gender, float moneySpent, int age, String typeOfCustomer, ArrayList<String> hobbies) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.gender = gender;
        this.moneySpent = moneySpent;
        this.age = age;
        this.typeOfCustomer = typeOfCustomer;
        this.hobbies = hobbies;
    }

    /**
     * @return the id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id the id to set
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return the firstName
     */
    public String getFirstName() {
        return firstName;
    }

    /**
     * @param firstName the firstName to set
     */
    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    /**
     * @return the lastName
     */
    public String getLastName() {
        return lastName;
    }

    /**
     * @param lastName the lastName to set
     */
    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    /**
     * @return the gender
     */
    public String getGender() {
        return gender;
    }

    /**
     * @param gender the gender to set
     */
    public void setGender(String gender) {
        this.gender = gender;
    }

    /**
     * @return the moneySpent
     */
    public float getMoneySpent() {
        return moneySpent;
    }

    /**
     * @param moneySpent the moneySpent to set
     */
    public void setMoneySpent(float moneySpent) {
        this.moneySpent = moneySpent;
    }

    /**
     * @return the age
     */
    public int getAge() {
        return age;
    }

    /**
     * @param age the age to set
     */
    public void setAge(int age) {
        this.age = age;
    }

    /**
     * @return the typeOfCustomer
     */
    public String getTypeOfCustomer() {
        return typeOfCustomer;
    }

    /**
     * @param typeOfCustomer the typeOfCustomer to set
     */
    public void setTypeOfCustomer(String typeOfCustomer) {
        this.typeOfCustomer = typeOfCustomer;
    }

    /**
     * @return the hobbies
     */
    public ArrayList<String> getHobbies() {
        return hobbies;
    }

    /**
     * @param hobbies the hobbies to set
     */
    public void setHobbies(ArrayList<String> hobbies) {
        this.hobbies = hobbies;
    }
    
    
    
}

File: Completo/ws21GUI/ws21GUISwing.py/ShoppingCenter/model/customer.py
class Customer:
    def __init__(self, id_customer, first_name, last_name, gender, money_spent, age, type_of_customer, hobbies):
        self._id = id_customer
        self._first_name = first_name
        self._last_name = last_name
        self._gender = gender
        self._money_spent = money_spent
        self._age = age
        self._type_of_customer = type_of_customer
        self._hobbies = hobbies

    def get_id(self): return self._id
    def set_id(self, value): self._id = value

    def get_first_name(self): return self._first_name
    def set_first_name(self, value): self._first_name = value

    def get_last_name(self): return self._last_name
    def set_last_name(self, value): self._last_name = value

    def get_gender(self): return self._gender
    def set_gender(self, value): self._gender = value

    def get_money_spent(self): return self._money_spent
    def set_money_spent(self, value): self._money_spent = value

    def get_age(self): return self._age
    def set_age(self, value): self._age = value

    def get_type_of_customer(self): return self._type_of_customer
    def set_type_of_customer(self, value): self._type_of_customer = value

    def get_hobbies(self): return self._hobbies
    def set_hobbies(self, value): self._hobbies = value

    def __str__(self):
        return f"Customer{{id={self._id}, firstName={self._first_name}, lastName={self._last_name}, gender={self._gender}, moneySpent={self._money_spent}, age={self._age}, typeOfCustomer={self._type_of_customer}, hobbies={self._hobbies}}}"